scapy/contrib/ikev2.py

```diff
@@ -664,6 +665,17 @@ class IKEv2_payload_Nonce(IKEv2_class):
 
 
 class IKEv2_payload_Notify(IKEv2_class):
+    @classmethod
+    def dispatch_hook(cls, _pkt=None, *args, **kargs):
+        IKEv2NotifyMessageClasses = {
+            16407: IKEv2_payload_Notify_Redirect,
+            16408: IKEv2_payload_Notify_Redirected_From
+        }
+
+        if _pkt and len(_pkt) >= 8:
+            return IKEv2NotifyMessageClasses.get(struct.unpack("!H", _pkt[6:8])[0], cls)
+        return cls
```

Comment:
@gpotter2 I'm aware now that this is not the optimal way to do it. So I will probably let this PR hang around until I finished refactoring the ikev module.

Reply:
#3925 has a much better solution.